ang/civiawards/shared/directives/contacts-group-dropdown.directive.js

```diff
@@ -0,0 +1,193 @@
+(function (angular, $, _) {
+  var module = angular.module('civiawards');
+
+  module.directive('civiawardContactsGroupDropdown', function (crmApi) {
+    return {
+      link: civiawardContactsGroupDropdownLink,
+      restrict: 'A',
+      require: 'ngModel'
+    };
+
+    /**
+     * Contacts Group Dropdown Link function
+     *
+     * @param {object} scope scope object of the controller
+     * @param {object} elem directives element
+     * @param {object} attr attributes of the directive element
+     * @param {object} ngModel ng model service
+     */
+    function civiawardContactsGroupDropdownLink (scope, elem, attr, ngModel) {
+      scope.groups = {
+        more: false,
+        results: [{
+          text: 'Include Group',
+          children: []
+        }, {
+          text: 'Exclude Group',
+          children: []
+        }]
+      };
+
+      (function init () {
+        ngModel.$render = refreshUI;
+
+        fetchGroupsData()
+          .then(prepareGroupsDataForSelect2)
+          .then(initSelect2)
+          .then(refreshUI);
+
+        scope.$watchCollection('recips.include', refreshUI);
+        scope.$watchCollection('recips.exclude', refreshUI);
+      }());
+
+      /**
+       * Initialise Select 2
+       */
+      function initSelect2 () {
+        elem.select2({
+          containerCssClass: 'civiaward__contacts-group__container',
+          multiple: true,
+          data: scope.groups,
+          dropdownAutoWidth: true,
+          formatResult: formatItem,
+          formatSelection: formatItem
+        });
+
+        elem.on('select2-selecting', selectEventHandler);
+
+        $(elem).on('select2-removing', removeEventHandler);
+      }
+
+      /**
+       * @param {Array} groupsData groups data
+       */
+      function prepareGroupsDataForSelect2 (groupsData) {
```

Comment:
missing description for this function

Reply:
Also, can we make it so it returns information instead of modifying a parameter?